lib/led-matrix/src/LEDMatrix.cpp

```diff
@@ -0,0 +1,78 @@
+/* LED Matrix Source File */
+
+#include "LEDMatrix.h"
+
+// Constructor: An LED Matrix is defined by the pins it is connected to
+LEDMatrix :: LEDMatrix(PinName R, PinName G, PinName B)
+    : _RChannel(R), _GChannel(G), _BChannel(B)
+{}
+
+// This function takes in values from 0 to 255 for each color stream and maps them to a PWM signal
+LEDMatrix :: void setColor(uint8_t R, uint8_t G, uint8_t B)
+{
+    _RChannel.pulsewidth((1.0/255.0)*R);
+    _GChannel.pulsewidth((1.0/255.0)*G);
+    _BChannel.pulsewidth((1.0/255.0)*B);
+}
+
+// This function is just for convenience. It simply calls the above function. For example if RED is passed in, call the above function with setColor(255, 0, 0)
+LEDMatrix :: void setColor(color c)
+{
+    switch(c)
+    {
+        case color::RED:
+            setColor(255, 0, 0);
+            break;
+        case color::GREEN:
+            setColor(0, 255, 0);
+            break;
+        case color::BLUE:
+            setColor(0, 0, 255);
+            break;
+    }
+}
+
+// This function should continously set the specified color, wait some time, turn off all LEDs, wait some time and repeat until another function is called by the main.cpp
```

Comment:
we decided this isnt continous right? ur comment says continuous. this just flashes n-times,where n is defined in the bridge repo